### PR TITLE
Beluga dev

### PR DESCRIPTION
**Pull request by TortillasAlfred**

### CODE REVIEW

File: beluga_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+numpy
```
Comment (davebulaval) [inline beluga_requirements.txt:1]: to be remove

File: generic_beluga_launcher.sh [hunk 2]
```diff
@@ -1,18 +1,25 @@
 #!/bin/bash
```
Comment (davebulaval) [inline generic_beluga_launcher.sh:1]: to be remove

File: src/domain/experiment.py [hunk 3]
```diff
@@ -26,13 +29,21 @@ def get_parameters_combinations(self):
...
+                                ablated_param: param_value
+                                for ablated_param, param_value in zip(ablated_params, params_combination)
+                            }
+                            run_params.update(new_params)
+                            yield run_params, self.get_sbatch_args(run_params)
                     else:
-                        yield run_params
+                        yield run_params, self.get_sbatch_args(run_params)
+
+    def get_sbatch_args(self, run_params):
```
Comment (davebulaval) [inline src/domain/experiment.py:44]: What is that ?

File: src/domain/experiment.py [hunk 4]
```diff
@@ -119,37 +125,50 @@ class ReweightAblationExperiment(OriginalExperiment):
...
     EXPERIMENT_NAME = 'other_languages_original'
-    LANGUAGE_PARAMS = {
-        'source_language': ['en'],
-        'target_language': ['et', 'fa', 'lv', 'vi']
+    LANGUAGE_PARAMS = {'source_language': ['en'], 'target_language': ['et', 'fa', 'lv', 'vi']}
+
+    def __init__(self, base_config):
+        super().__init__(base_config)
+
+    def get_sbatch_args(self, run_params):
```
Comment (davebulaval) [inline src/domain/experiment.py:158]: What is that ?

File: src/domain/experiment.py [hunk 5]
```diff
@@ -119,37 +125,50 @@ class ReweightAblationExperiment(OriginalExperiment):
...
+    EXPERIMENT_NAME = 'other_languages_stochastic_ablation'
+    LANGUAGE_PARAMS = {'source_language': ['en'], 'target_language': ['et', 'fa', 'lv', 'vi']}
+    CHANGING_PARAMS = {
+        'stochastic_initial': [1.0],
     }
+
     def __init__(self, base_config):
         super().__init__(base_config)
 
+    def get_sbatch_args(self, run_params):
```
Comment (davebulaval) [inline src/domain/experiment.py:172]: What is that ?

File: src/domain/experiment.py [hunk 6]
```diff
@@ -207,29 +226,121 @@ class OtherLanguagesRandomCutoffSeedDictionaryAblationExperiment(RandomCutoffSee
         'source_language': ['en'],
         'target_language': ['et', 'fa', 'lv', 'vi']
     }
+
+    def __init__(self, base_config):
+        super().__init__(base_config)
+
+    def get_sbatch_args(self, run_params):
```
Comment (davebulaval) [inline src/domain/experiment.py:241]: What is that ?

File: src/domain/experiment.py [hunk 7]
```diff
@@ -207,29 +226,121 @@ class OtherLanguagesRandomCutoffSeedDictionaryAblationExperiment(RandomCutoffSee
...
+
+class OtherLanguagesRandomSeedDictionaryAblationExperiment(OriginalExperiment):
+    EXPERIMENT_NAME = 'other_languages_random_seed_dictionary_ablation'
+    LANGUAGE_PARAMS = {'source_language': ['en'], 'target_language': ['et', 'fa', 'lv', 'vi']}
+    CHANGING_PARAMS = {'seed_dictionary_method': ['random_raw']}
+
+    def __init__(self, base_config):
+        super().__init__(base_config)
+
+    def get_sbatch_args(self, run_params):
```
Comment (davebulaval) [inline src/domain/experiment.py:253]: What is that ?

File: src/domain/experiment.py [hunk 8]
```diff
@@ -207,29 +226,121 @@ class OtherLanguagesRandomCutoffSeedDictionaryAblationExperiment(RandomCutoffSee
...
+        return {'time': '0-1:00'}
+
+
+class OtherLanguagesRandomCutoffSeedDictionaryAblationExperiment(OriginalExperiment):
+    EXPERIMENT_NAME = 'other_languages_random_cutoff_seed_dictionary_ablation'
```
Comment (davebulaval) [inline src/domain/experiment.py:258]: why not `__name__ = other_languages_random_cutoff_seed_dictionary_ablation `

File: src/domain/experiment.py [hunk 9]
```diff
@@ -207,29 +226,121 @@ class OtherLanguagesRandomCutoffSeedDictionaryAblationExperiment(RandomCutoffSee
...
+        return {'time': '0-1:00'}
+
+
+class GridSearchExperiment(OriginalExperiment):
+    def __init__(self, base_config):
+        super().__init__(base_config)
+
+    def __get_run_params(self, run):
+        if len(self.CHANGING_PARAMS) == 1:
+            return run.data.params[list(self.CHANGING_PARAMS)[0]]
```
Comment (davebulaval) [inline src/domain/experiment.py:275]: Why not list already `CHANGING_PARAMS`

File: src/domain/experiment.py [hunk 10]
```diff
@@ -207,29 +226,121 @@ class OtherLanguagesRandomCutoffSeedDictionaryAblationExperiment(RandomCutoffSee
...
+
+    def aggregate_runs(self):
+        runs = self.get_runs()
+        accuracies = defaultdict(lambda: defaultdict(list))
+        times = defaultdict(lambda: defaultdict(list))
+        num_iters = defaultdict(lambda: defaultdict(list))
+        avg_iter_times = defaultdict(lambda: defaultdict(list))
+        for run in runs:
+            if self._is_a_valid_run(run):
+                minutes = max(((run.info.end_time - run.info.start_time) // 60 // 60) % 60, 1)
```
Comment (davebulaval) [inline src/domain/experiment.py:287]: Why not use datetime object ?

File: src/domain/experiment.py [hunk 11]
```diff
@@ -207,29 +226,121 @@ class OtherLanguagesRandomCutoffSeedDictionaryAblationExperiment(RandomCutoffSee
...
-            'fa',
-            'fi',
-            'it'
-            'lv',
-            'vi'
-        ]
-    }
+    LANGUAGE_PARAMS = {'source_language': ['en'], 'target_language': ['de', 'es', 'et', 'fa', 'fi', 'it' 'lv', 'vi']}
 
     # TODO Change a bunch of params here
```
Comment (davebulaval) [inline src/domain/experiment.py:350]: yep!

File: src/scripts/generate_tables.py [hunk 12]
```diff
@@ -15,20 +15,28 @@ def generate_table_1(configs):
```
Comment (davebulaval) [inline src/scripts/generate_tables.py:16]: `method` in doc should be function instead.

File: src/scripts/runner.py [hunk 13]
```diff
@@ -71,13 +69,14 @@ def __init__(self, run_launcher, num_runs, cuda):
     def run_experiment_for_table(self, table):
         for name, experiment in table.get_experiments():
             logging.info("Running experiment: {}".format(experiment.EXPERIMENT_NAME))
-            for config in experiment.get_parameters_combinations():
+            for config, sbatch_args in experiment.get_parameters_combinations():
```
Comment (davebulaval) [inline src/scripts/runner.py:71]: `sbatch_args` ?

File: beluga_requirements.txt (hunk 1 above)
Comment (TortillasAlfred) [reply beluga_requirements.txt:1]: Agreed

File: generic_beluga_launcher.sh (hunk 2 above)
Comment (TortillasAlfred) [reply generic_beluga_launcher.sh:1]: Agreed

File: src/domain/experiment.py (hunk 3 above)
Comment (TortillasAlfred) [reply src/domain/experiment.py:44]: Beluga related stuff, to be removed

File: src/domain/experiment.py (hunk 4 above)
Comment (TortillasAlfred) [reply src/domain/experiment.py:158]: Beluga related stuff, to be removed

File: src/domain/experiment.py (hunk 5 above)
Comment (TortillasAlfred) [reply src/domain/experiment.py:172]: Beluga related stuff, to be removed.

File: src/domain/experiment.py (hunk 9 above)
Comment (TortillasAlfred) [reply src/domain/experiment.py:275]: CHANGING_PARAMS is a dict, I only want to read the params for all changing params, where in this case it is only a single key.

File: src/domain/experiment.py (hunk 10 above)
Comment (TortillasAlfred) [reply src/domain/experiment.py:287]: I only added the max with 1 to ensure no zero division error

File: src/scripts/runner.py (hunk 13 above)
Comment (TortillasAlfred) [reply src/scripts/runner.py:71]: Will be removed in master

File: src/domain/experiment.py (hunk 10 above)
Comment (davebulaval) [reply src/domain/experiment.py:287]: Yeah, but this logic of changing into minutes is already managed by the class datetime.

File: src/scripts/runner.py (hunk 13 above)
Comment (davebulaval) [reply src/scripts/runner.py:71]: Ok

File: src/scripts/generate_tables.py (hunk 12 above)
Comment (davebulaval) [reply src/scripts/generate_tables.py:16]: 02673fa

**Review by davebulaval — APPROVED** [review]

**Pull request merged**